### PR TITLE
Migrate InstagramPhotosExporter.

### PR DESCRIPTION
**Pull request by rtannenbaum**
Not tested yet.

#285

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java [hunk 1]
```diff
@@ -15,24 +15,110 @@
...
     Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  private static final String FAKE_ALBUM_ID = "instagramAlbum";
+
+  private final HttpTransport httpTransport;
+
+  public InstagramPhotoExporter() {
+    try {
+      this.httpTransport = GoogleNetHttpTransport.newTrustedTransport();
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: I think we should pass the HttpTransport thru in the constructor so the InstagramTransferExtension can obtain it from the context and pass it through. 

That way we can provide global HttpTransport objects.

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java [hunk 2]
```diff
@@ -15,24 +15,110 @@
...
+import org.dataportabilityproject.transfer.instagram.photos.model.MediaFeedData;
+import org.dataportabilityproject.transfer.instagram.photos.model.MediaResponse;
 import org.dataportabilityproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.dataportabilityproject.types.transfer.models.photos.PhotoAlbum;
+import org.dataportabilityproject.types.transfer.models.photos.PhotoModel;
 import org.dataportabilityproject.types.transfer.models.photos.PhotosContainerResource;
 
 public class InstagramPhotoExporter implements
     Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+  private static final ObjectMapper MAPPER =
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: I think we should pass the ObjectMapper and/or TypeManager thru the constructor and have it be obtained or created by the InstragramTransferExtension from the context.

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Caption.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.instagram.photos.model;
+
+/** DataModel for a Caption object in the Instagram API. Instantiated by JSON mapping. */
+public final class Caption {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Caption.java]: Will these need JSON mappings?

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/ImageData.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.instagram.photos.model;
+
+/** DataModel for a single image object in the Instagram API. Instantiated by JSON mapping. */
+public final class ImageData {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/ImageData.java]: Will these need JSON mappings?

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaFeedData.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.instagram.photos.model;
+
+/** DataModel for a media feed in the Instagram API. Instantiated by JSON mapping. */
+public final class MediaFeedData {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaFeedData.java]: Will these need JSON mappings?

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaResponse.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.instagram.photos.model;
+
+import java.util.List;
+
+/** DataModel for the result of a media query in the Instagram API. Instantiated by JSON mapping. */
+public final class MediaResponse {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaResponse.java]: Will these need JSON mappings?

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java [hunk 7]
```diff
@@ -15,24 +15,110 @@
...
-    return null; // TODO: implement
+    return exportPhotos(authData, Optional.of(exportInformation.getPaginationData()));
+  }
+
+  private ExportResult<PhotosContainerResource> exportPhotos(TokensAndUrlAuthData authData,
+      Optional<PaginationData> pageData) {
+    Preconditions.checkNotNull(authData);
+    MediaResponse response;
+    try {
+      response = makeRequest("https://api.instagram.com/v1/users/self/media/recent",
```
Comment (seehamrun) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: make the url a constant

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Meta.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.instagram.photos.model;
+
+/** DataModel for a metadata about a response in the Instagram API. Instantiated by JSON mapping. */
+public final class Meta {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Meta.java]: Will these need JSON mappings?

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java (hunk 2 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: Good point, done

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java (hunk 1 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: done

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java (hunk 7 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/InstagramPhotoExporter.java]: done

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Caption.java (hunk 3 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Caption.java]: As discussed offline - these were working in the old core code and I ported them over verbatim, so I think they should still work. They must be using default mappings (except a couple in one class that have custom mappings)

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/ImageData.java (hunk 4 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/ImageData.java]: see other JSON mapping reply

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaFeedData.java (hunk 5 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaFeedData.java]: see other JSON mapping reply

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaResponse.java (hunk 6 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/MediaResponse.java]: see other JSON mapping reply

File: extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Meta.java (hunk 8 above)
Comment (rtannenbaum) [reply extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/dataportabilityproject/transfer/instagram/photos/model/Meta.java]: see other JSON mapping reply

**Pull request merged**